src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/BuildTest.cs

```diff
@@ -1058,9 +1058,11 @@ public void BuildApplicationWithJavaSource (bool isRelease, bool expectedResult)
 						Assert.AreEqual (runtime.Size, inApkRuntime.Size, "expected {0} got {1}", expectedRuntime, inApkRuntime.Runtime);
 						inApk = ZipHelper.ReadFileFromZip (apk, string.Format ("lib/{0}/libmono-profiler-log.so", abi));
 						if (string.Compare (expectedRuntime, "debug", StringComparison.OrdinalIgnoreCase) == 0) {
-							Assert.IsNotNull (inApk, "libmono-profiler-log.so should exist in the apk.");
+							if (inApk == null)
+								Assert.Fail ("libmono-profiler-log.so should exist in the apk.");
```

Comment:
Why replace `Assert.IsNotNull` with `if (inapt != null) Assert.Fail()`? Aren't those semantically the same?

Reply:
I cherry-picked that from the abandoned https://github.com/xamarin/xamarin-android/pull/723 so I don't have a lot of context, but the commit message says this: `Avoid dumping the content of libmono-profiler-log.so when test is falling` which makes sense to me since `Assert.IsNotNull` would dump the value of `inApk` to the log.

Reply:
Ah. Good reason. :-)